pennylane/gradients/pulse_gradient.py

```diff
@@ -0,0 +1,532 @@
+        the coefficient functions, :math:`\partial f_j / \partial v_k`, are computed classically,
+        and :math:`\widetilde{C_j}` is a linear combination of the results from modified pulse
+        sequence executions based on generalized parameter-shift rules
+        (see e.g. `Kyriienko and Elfving (2022) <https://arxiv.org/abs/2108.01218>`_ or
+        `Wierichs et al. (2022) <https://doi.org/10.22331/q-2022-03-30-677>`_ for more details
+        and :func:`~.param_shift` for an implementation of the non-stochastic generalized shift
+        rules)
+        Given the parameter shift rule with coefficients :math:`\{y_\ell\}` and shifts
+        :math:`\{x_\ell\}` for the single-parameter pulse :math:`\exp(-i \theta H_j)`,
+        the linear combination is given by
+
+        .. math::
+
+            \widetilde{C_j}(\boldsymbol{v}, \tau)&=\sum_{\ell=1} y_\ell
+            \bra{\psi_{j}(\boldsymbol{v}, x_\ell, \tau)} B
+            \ket{\psi_{j}(\boldsymbol{v}, x_\ell, \tau)} \\
+            \ket{\psi_{j}(\boldsymbol{v}, x_\ell, \tau)}
+            &= U_{\boldsymbol{v}}(T, \tau) e^{-i x_\ell H_j}
+            U_{\boldsymbol{v}}(\tau, 0)\ket{\psi_0}.
+
+        In practice, the time integral over :math:`\tau` is computed by sampling values for
+        the time, evaluating the integrand, and averaging appropriately. The probability
+        distribution used for the sampling may have a significant impact on the quality of the
+        obtained estimates, in particular with regards to their variance.
+        In this function, a uniform distribution over the interval :math:`[0, t]` is used,
+        which often can be improved upon.
+
+        **Examples**
+
+        Consider the pulse generated by
+
+        .. math::
+
+            H(\boldsymbol{v}, t) = \frac{1}{2} X\otimes I + v_1 Z\otimes Z + \sin(v_2 t) I\otimes X
+
+        and the observable :math:`B=I\otimes Y`. There are two variational parameters, :math:`v_1`
+        and :math:`v_2`, for which we may compute the derivative of the cost function:
+
+        .. math::
+
+            \frac{\partial C}{\partial v_1}
+            &= \int_{0}^{T} \mathrm{d}\tau \ \widetilde{C_1}((v_1, v_2), \tau)\\
+            \frac{\partial C}{\partial v_2}
+            &= \int_{0}^{T} \mathrm{d}\tau \cos(v_2 \tau) \tau \ \widetilde{C_2}((v_1, v_2), \tau)\\
+            \widetilde{C_j}((v_1, v_2), \tau)&=
+            \bra{\psi_{j}((v_1, v_2), \pi/4, \tau)} B
+            \ket{\psi_{j}((v_1, v_2), \pi/4, \tau)}\\
+            &-\bra{\psi_{j}((v_1, v_2), -\pi/4, \tau)} B
+            \ket{\psi_{j}((v_1, v_2), -\pi/4, \tau)} \\
+            \ket{\psi_{j}((v_1, v_2), x, \tau)}
+            &= U_{(v_1, v_2)}(T, \tau) e^{-i x H_j}U_{(v_1, v_2)}(\tau, 0)\ket{0}.
+
+        Here we used the partial derivatives
+
+        .. math::
+
+            \frac{\partial f_1}{\partial v_1}((v_1, v_2), t) &= 1\\
+            \frac{\partial f_1}{\partial v_1}((v_1, v_2), t) &= \cos(v_2 t) t \\
```

Comment:
```suggestion
            \frac{\partial f_2}{\partial v_2}((v_1, v_2), t) &= \cos(v_2 t) t \\
```

Reply:
Consider to remove all the $((v_1, v_2), t)$ arguments here

Reply:
You mean in this block of equations or more generally? :) I wanted to keep it in analogy to the general case where we carry the v around, and make the dependencies explicit :thinking:

Reply:
Just meant in this block :) I like it in the updated version a lot 👍